src/realm/parser/driver.cpp

```diff
@@ -797,12 +797,49 @@ Query GeoWithinNode::visit(ParserDriver* drv)
     else {
         size_t arg_no = size_t(strtol(argument.substr(1).c_str(), nullptr, 10));
         auto right_type = drv->m_args.is_argument_null(arg_no) ? DataType(-1) : drv->m_args.type_for_argument(arg_no);
-        if (right_type != type_Geospatial) {
+
+        Geospatial geo;
```

Comment:
Hides member variable `geo`. Confusing. Perhaps the member should be `geo_node` or `geospatial`,

Reply:
Good eye, that is confusing. Updated.